src/routes/transactions/mappers/transaction-preview.mapper.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@nestjs/common';
+import * as winston from 'winston';
+import { MultisigTransaction } from '../../../domain/safe/entities/multisig-transaction.entity';
+import { Safe } from '../../../domain/safe/entities/safe.entity';
+import { DataDecodedService } from '../../data-decode/data-decoded.service';
+import { GetDataDecodedDto } from '../../data-decode/entities/get-data-decoded.dto.entity';
+import { PreviewTransactionDto } from '../entities/preview-transaction.dto.entity';
+import { TransactionPreview } from '../entities/transaction-preview.entity';
+import { TransactionDataMapper } from './common/transaction-data.mapper';
+import { MultisigTransactionInfoMapper } from './common/transaction-info.mapper';
+
+@Injectable()
+export class TransactionPreviewMapper {
+  constructor(
+    private readonly transactionInfoMapper: MultisigTransactionInfoMapper,
+    private readonly transactionDataMapper: TransactionDataMapper,
+    private readonly dataDecodedService: DataDecodedService,
```

Comment:
Yes, I agree, I'll move the shared logic from `DataDecodedService` to the domain (to `DataDecodedRepository`).

Reply:
In fact, I've removed the additional logic that `DataDecodedService` had, since it was just redundant mapping logic, the domain and route objects map 1:1 in this case: [e1b9fb3](https://github.com/5afe/safe-client-gateway-nest/pull/291/commits/e1b9fb3c1c746a4879855e3b7ad35002eb9dd716)